File: app/src/main/java/at/shockbytes/corey/body/points/WeightPoint.java
package at.shockbytes.corey.body.points;

import io.realm.RealmObject;

public class WeightPoint extends RealmObject implements Comparable<WeightPoint> {

    private long time;
    private double weight;

    public WeightPoint() {
        this(0, 0);
    }

    public WeightPoint(long time, double weight) {
        this.time = time;
        this.weight = weight;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }


    @Override
    public int compareTo(WeightPoint weightPoint) {
        return (int) (weight - weightPoint.weight);
    }
}

File: app/src/main/java/at/shockbytes/corey/body/info/WeightPoint.kt
package at.shockbytes.corey.body.info

data class WeightPoint(val timeStamp: Long = 0, val weight: Double = 0.0): Comparable<WeightPoint> {

    override fun compareTo(other: WeightPoint): Int {
        return (timeStamp - other.timeStamp).toInt()
    }
}
